src/dra/6D59C.c

```diff
@@ -1478,7 +1478,56 @@ bool CheckQuarterCircleForwardInput(void) {
     return 0;
 }
 
-INCLUDE_ASM("dra/nonmatchings/6D59C", func_8011081C);
+bool CheckBackForwardInput(void) {
+    s32 directionsPressed;
+    s32 backward;
+    s32 forward;
+
+    directionsPressed =
+        g_Player.padPressed & (PAD_UP | PAD_RIGHT | PAD_DOWN | PAD_LEFT);
+    if (!g_WasFacingLeft2) {
+        forward = PAD_RIGHT;
+    } else {
+        forward = PAD_LEFT;
+    }
+    switch (g_ButtonCombo[COMBO_BF].buttonsCorrect) {
+    case 0:
+        g_WasFacingLeft2 = PLAYER.facingLeft;
+        if (!g_WasFacingLeft2) {
+            backward = PAD_LEFT;
+        } else {
+            backward = PAD_RIGHT;
+        }
+        if (directionsPressed == backward) {
+            g_ButtonCombo[COMBO_BF].timer = 15;
+            g_ButtonCombo[COMBO_BF].buttonsCorrect++;
+        } else if (--g_ButtonCombo[COMBO_BF].timer == 0) {
+            g_ButtonCombo[COMBO_BF].buttonsCorrect = 0;
+        }
+        break;
+    case 1:
+        if (directionsPressed == forward) {
+            g_ButtonCombo[COMBO_BF].timer = 15;
+            g_ButtonCombo[COMBO_BF].buttonsCorrect = 0xFF;
```

Comment:
I'm thinking we should define 0xFF as `COMBO_COMPLETE` or similar

Reply:
Added one, I just put it in this file for now since this file seems to host everything relating to those functions but we can move it to dra.h, either if you want it now, or if we find external uses?

Reply:
There's also a usage here https://github.com/Xeeynamo/sotn-decomp/blob/141c8b22c726f8dec8db8863adc529263cb2c7a7/src/dra/75F54.c#L2914

Probably dra.h makes the most sense?

Reply:
Good find, adding to dra.h alongside the ButtonComboState definition, seems best to group things that are related in usage